Tasks/CopyFilesToMachines/CopyFilesToMachines.ps1

```diff
@@ -160,7 +109,7 @@ else
 	
         Write-Output "Copy will be Started for - $machine"
 
-        $job = Start-Job -ScriptBlock $CopyJob -ArgumentList $environmentName, $envOperationId, $machine, $sourcePath, $targetPath, $machineUserName, $machinePassword, $cleanTargetBeforeCopy
+        $job = Start-Job -ScriptBlock $CopyJob -ArgumentList $environmentName, $envOperationId, $machine, $sourcePath, $targetPath, $machineUserName, $machinePassword, $cleanTargetBeforeCopy, $connection
 
         $Jobs.Add($job.Id, $machine)
     
```

Comment:
the logic of calling Get-Job twice is, first call check if there is any job is there and if so we give 10 sec time to complete that job and do a Get-Job again. Then check the Job status for the jobs.